napari/layers/points/points.py

```diff
@@ -851,23 +853,37 @@ def current_size(self) -> Union[int, float]:
 
     @current_size.setter
     def current_size(self, size: Union[None, float]) -> None:
-        if (isinstance(size, numbers.Number) and size < 0) or (
-            isinstance(size, list) and min(size) < 0
-        ):
+        if isinstance(size, (list, tuple, np.ndarray)):
             warnings.warn(
-                message=trans._(
-                    'current_size value must be positive, value will be left at {value}.',
+                trans._(
+                    "Point sizes must be isotropic; the average from each dimension will be used instead. "
+                    "This will become an error in a future version.",
+                    deferred=True,
+                ),
+                category=DeprecationWarning,
+                stacklevel=2,
+            )
+            size = size[-1]
+        if not isinstance(size, numbers.Number):
+            raise TypeError(
+                trans._(
+                    'currrent size must be a number',
+                    deferred=True,
+                )
+            )
+        if size < 0:
+            raise ValueError(
+                trans._(
+                    'current_size value must be positive.',
                     deferred=True,
-                    value=self.current_size,
                 ),
-                category=RuntimeWarning,
             )
-            size = self.current_size
+
         self._current_size = size
         if self._update_properties and len(self.selected_data) > 0:
-            for i in self.selected_data:
-                self.size[i, :] = (self.size[i, :] > 0) * size
-            self._clear_extent_augmented()
+            idx = np.fromiter(self.selected_data, dtype=int)
+            # TODO: explain why this check; only if size > 0?
```

Comment:
Is this a TODO for you or for someone in the future? I can't explain the need for the check either, so I'd also be down to simplify it to `self.size[idx] *= size` and remove the TODO.

Reply:
I added it because I didn't understand this check but it was already there, so I left it for the future. However, I now played around a bit, and I'm pretty sure it's wrong (we definitely want to allow points to have size 0, and nothing seems to break on the vispy side when doing so... I'm running tests now).